### PR TITLE
Improve typing in pg dbops

### PR DESCRIPTION
**Pull request by aljazerzen**
I added typing annotations while debugging and trying to understand the code.

### CODE REVIEW

**Comment by elprans** [comment]
Did you run these past mypy?

**Comment by aljazerzen** [comment]
Yes, it doesn't report errors. And it should start typechecking functions as soon as they get type annotations, right?

**Comment by elprans** [comment]
Our default mypy config is pretty weak and might ignore errors.  My advice would be to add the newly-typed modules to one of stricter `tool.mypy.overrides` in `pyproject.toml` for more thorough checking.

File: edb/pgsql/dbops/functions.py [hunk 1]
```diff
@@ -168,9 +191,15 @@ def code(self, block: base.PLBlock) -> str:
...
 class DropFunction(ddl.DDLOperation, FunctionOperation):
     def __init__(
-            self, name, args, *,
-            if_exists=False,
-            has_variadic=False, conditions=None, neg_conditions=None):
+        self,
+        name: Tuple[str, ...],
+        args: Sequence[str | Tuple[str, str, str]],
+        *,
+        if_exists=False,
```
Comment (msullivan) [inline edb/pgsql/dbops/functions.py:198]: `: bool`?

File: edb/pgsql/dbops/tables.py [hunk 2]
```diff
@@ -327,7 +328,11 @@ def code(self, block: base.PLBlock) -> str:
 
 
 class AlterTableBaseMixin:
-    def __init__(self, name, contained=False, **kwargs):
+
+    name: Tuple[str, ...]
+    contained: bool
+
+    def __init__(self, name: Tuple[str, ...], contained=False, **kwargs):
```
Comment (msullivan) [inline edb/pgsql/dbops/tables.py:335]: `: bool`?

**Review by msullivan — APPROVED** [review]
Sure. Agree with Elvis that it is probably worth trying making some stuff stricter in pyproject.toml`

The usefulness of this will be pretty limited until (unless) we enable errors in `pgsql/delta.py`... And that's gonna be tough for a few reasons (one I think being that a bunch of dodgy python magic happens)

**Pull request merged**